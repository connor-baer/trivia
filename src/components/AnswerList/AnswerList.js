import React from 'react';
import PropTypes from 'prop-types';

import Answer from './components/Answer';
import styled, { css } from 'react-emotion';

const containerStyles = ({ theme }) => css`
  display: flex;
  flex-wrap: wrap;
`;

const Container = styled('div')(containerStyles);

const AnswerList = ({ answers, onSelect, selected }) => {
  return (
    <Container>
      {answers &&
        answers.map(({ option, id, className, ...props }) => (
          <Answer
            key={id}
            value={id}
            name="answer-list"
            onToggle={onSelect}
            checked={id === selected}
            aria-labelledby="question"
            role="radiogroup"
          >
            {option}
          </Answer>
        ))}
    </Container>
  );
};

AnswerList.propTypes = {
  answers: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.number.isRequired,
      label: PropTypes.string.isRequired
    })
  ),
  onSelect: PropTypes.func.isRequired,
  selected: PropTypes.number.isRequired
};

AnswerList.defaultProps = {
  answers: []
};

/**
 * @component
 */
export default AnswerList;
