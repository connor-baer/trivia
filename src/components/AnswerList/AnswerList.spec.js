import React from 'react';

import AnswerList from './AnswerList';

const defaultProps = {
  answers: [
    {
      id: '1',
      option: 'Foo'
    },
    {
      id: '2',
      option: 'Bar'
    }
  ],
  onSelect: jest.fn()
};

const renderAnswerList = (props = {}) =>
  render(<AnswerList {...{ ...defaultProps, ...props }} />);

describe('AnswerList', () => {
  it('should render with default styles', () => {
    const { container } = renderAnswerList();
    expect(container.firstChild).toMatchSnapshot();
  });

  it.skip('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<AnswerList {...defaultProps} />);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
