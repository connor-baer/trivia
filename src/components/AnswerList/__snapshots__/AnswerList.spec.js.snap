// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerList should render with default styles 1`] = `
.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-0 {
  border: 0px;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0px;
  position: absolute;
  white-space: nowrap;
  width: 1px;
  width: 50%;
  padding: 0 12px 12px 0;
  cursor: pointer;
}

.emotion-0:focus + label::before {
  border-width: 2px;
  border-color: #3388FF;
}

.emotion-0::before,
.emotion-0::after {
  content: none;
}

.emotion-2 {
  color: #5C656F;
  padding-left: 24px;
  position: relative;
  width: 50%;
  padding: 0 12px 12px 0;
  cursor: pointer;
}

.emotion-2::before {
  box-sizing: border-box;
  height: 16px;
  width: 16px;
  box-shadow: inset 0 1px 2px 0 rgba(102,113,123,0.12);
  background-color: #FFFFFF;
  border: 1px solid #9DA7B1;
  border-radius: 100%;
  content: '';
  display: block;
  position: absolute;
  top: 50%;
  left: 0;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition: border 0.05s ease-in;
  transition: border 0.05s ease-in;
}

.emotion-2::after {
  box-sizing: border-box;
  height: 8px;
  width: 8px;
  background-color: #3388FF;
  border-radius: 100%;
  content: '';
  display: block;
  position: absolute;
  top: 50%;
  left: 4px;
  -webkit-transform: translateY(-50%) scale(0,0);
  -ms-transform: translateY(-50%) scale(0,0);
  transform: translateY(-50%) scale(0,0);
  opacity: 0;
  -webkit-transition: -webkit-transform 0.05s ease-in,opacity 0.05s ease-in;
  -webkit-transition: transform 0.05s ease-in,opacity 0.05s ease-in;
  transition: transform 0.05s ease-in,opacity 0.05s ease-in;
}

.emotion-2::before {
  border-color: #3388FF;
}

.emotion-2::after {
  -webkit-transform: translateY(-50%) scale(1,1);
  -ms-transform: translateY(-50%) scale(1,1);
  transform: translateY(-50%) scale(1,1);
  opacity: 1;
}

.emotion-2::before,
.emotion-2::after {
  content: none;
}

<div
  class="emotion-8"
>
  <input
    aria-labelledby="question"
    checked=""
    class="emotion-0 emotion-1"
    id="radio-button_1"
    name="answer-list"
    role="radiogroup"
    type="radio"
    value=""
  />
  <label
    aria-labelledby="question"
    class="emotion-2 emotion-3"
    for="radio-button_1"
    name="answer-list"
    role="radiogroup"
  />
  <input
    aria-labelledby="question"
    checked=""
    class="emotion-0 emotion-1"
    id="radio-button_2"
    name="answer-list"
    role="radiogroup"
    type="radio"
    value=""
  />
  <label
    aria-labelledby="question"
    class="emotion-2 emotion-3"
    for="radio-button_2"
    name="answer-list"
    role="radiogroup"
  />
</div>
`;
